statsfs/otelstats/subsys_metric_parser.go

```diff
@@ -0,0 +1,199 @@
+package otelstats
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Given a path to a statsfs file, return the metricName.
+// path = {m.StatsfsPath}/{m.SubSystemName}/{device paths(if any)}/{metricFileName}
+// metricName = {m.SubSystemName}/{metricFilename}
+// Example:
+//	Input:
+//		path = /sys/kernel/stats/net/eth0/latency
+//			where
+//				m.StatsfsPath = /sys/kernel/stats
+//				m.SubsystemName = net
+//				device paths = eth0
+//				metricFileName = latency
+//	Output:
+//		metricName = net/latency
+func (m SubsysMetrics) getMetricName(path string) string {
+	segs := strings.Split(path, "/")
+	metricFileName := segs[len(segs)-1]
+	metricName := strings.Join([]string{m.SubSystemName, metricFileName}, "/")
+	return metricName
+}
+
+// Given a path to a statsfs file, return the label of the metric.
+// path = {m.SubSystemPath}/{device paths(if any)}/{metricFileName}
+// label = {device paths (if any)}
+// Example:
+//	Input:
+//		path = /sys/kernel/stats/net/eth0/latency
+//			where
+//				m.StatsfsPath = /sys/kernel/stats
+//				m.SubSystemName = net
+//				m.SubSystemPath = /sys/kernel/stats/net
+//				device paths = /eth0/sub0
+//				metricFileName = latency
+//	Output:
+// 		label = /eth0/sub0
+func (m SubsysMetrics) getMetricLabel(path string) string {
+	metricStr := strings.Split(path, m.SubSystemPath)[1]
+	labelSeg := strings.Split(metricStr, "/")
+	label := strings.Join(labelSeg[:len(labelSeg)-1], "/")
+	return label
+}
+
+// Given a path to a statsfs file, return a MetricInfo struct with label
+// computed by getMetricLabel method and Path being the input path
+// (the path from where the metric could be retrieved)
+// Example:
+//	Input:
+//		m.SubSystemPath = /sys/kernel/stats/net
+//		path = /sys/kernel/stats/net/eth0/sub0/latency
+//	Output:
+//		MetricInfo{
+//			Label: /eth0/sub0
+//			Path: /sys/kernel/stats/net/eth0/sub0/latency
+//		}
+func (m SubsysMetrics) getMetricInfo(path string) MetricInfo {
+	label := m.getMetricLabel(path)
+	metricInfo := MetricInfo{Label: label, Path: path}
+	return metricInfo
+}
```

Comment:
I think all these comments and small functions cause more trouble when reading. I would suggest to have the MetricInfo contain `Name`, `Label` and `Path`.

And we have the function `getMetricInfo` to fill all of them in, with detailed comments about Input and Output. You can still keep those small functions, like `getMetricName`, `getMetricLabel`, but they don't need to be commented. Otherwise, it would be too distracting.